### PR TITLE
[easy] small improvements to indirectfile package

### PR DESCRIPTION
**Pull request by ttung**
1. Improve documentation.
2. Properly type the `convert` method.

Test plan: `make -j lint mypy`

### CODE REVIEW

**Comment by ttung** [comment]
Documentation improvements that you didn't even have to ask for, @ambrosejcarr!

**Comment by ambrosejcarr** [comment]
🥇

File: starfish/util/indirectfile/_base.py [hunk 1]
```diff
@@ -6,8 +6,14 @@
...
 
 class ConversionRecipe(Generic[RecipeResultType]):
+    """A conversion recipe represents a plausible contract to convert a string value to another data
+    type.  Each conversion recipe implements two methods -- a lightweight method to determine if
+    this recipe can be applied given a string value, and a load method that actually does the
+    conversion.
+    """
     @abc.abstractmethod
     def applicable(self, input_parameter: str) -> bool:
+        """Returns true iff this recipe might work for the given input."""
```
Comment (shanaxel42) [inline starfish/util/indirectfile/_base.py]:
```suggestion
        """Returns true if this recipe might work for the given input."""
```

Comment (ttung) [reply starfish/util/indirectfile/_base.py]: iff = if and only if.

**Pull request merged**